pkg/controller/controller_utils.go

```diff
@@ -756,8 +762,14 @@ func (s ActivePods) Less(i, j int) bool {
 		}
 	}
 	// 5. Pods with containers with higher restart counts < lower restart counts
-	if maxContainerRestarts(s[i]) != maxContainerRestarts(s[j]) {
-		return maxContainerRestarts(s[i]) > maxContainerRestarts(s[j])
+	regularRestartsI, sidecarRestartsI := maxContainerRestarts(s[i])
```

Comment:
note: these repeats are horrible, we should consider DRYing it one day (maybe with generics)